pkg/provisioner/ironic/ironic.go

```diff
@@ -342,7 +355,12 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged, force b
 
 	ironicNode, err = p.findExistingHost()
 	if err != nil {
-		result, err = transientError(errors.Wrap(err, "failed to find existing host"))
+		switch err.(type) {
+		case *macAddressConflictError:
```

Comment:
It would be nice here to use the error manipulations functions like `errors.As()` to check the error type

Reply:
I think switching on the error type is the common way of doing this in this project.

Reply:
That's right, I suggested that because since 1.13 the new error wrapping/manipulation functions have been introduced (see [1]). That's something anyhow that we could review more generally in another PR in case.

[1] https://golang.org/doc/go1.13#error_wrapping
[2] https://golang.org/pkg/errors/